docs/src/design/changing_the_primal.md

```diff
@@ -469,6 +469,15 @@ We don't have this in ChainRules.jl yet, because Julia is missing some definitio
 We have been promised them for Julia v1.7 though.
 You can see what the code would look like in [PR #302](https://github.com/JuliaDiff/ChainRules.jl/pull/302).
 
+## What things can be pulled out of the pullback?
```

Comment:
```suggestion
## What things can be taken out of the pullback?
```
Seems clearer?

Reply:
My understanding is that the motivation is that we can reuse the work that was done in the primal to reduce the work that needs to be done in the pullback.

But this current paragraph insinuates (at least to me) that: `if there is an operation you can do in the augmented primal, do it there, rather than in the pullback.` Is this true? I can imagine this is true if pullback gets called more than once, but that does not happen, right?

Reply:
The pullback gets called several times by `jacobian`.

It also sometimes does not get called at all, which should happen when the gradient is Zero (but doesn't always?) and may also happen because AD has called `rrule` on code it ought to know cannot have a derivative (as in https://github.com/FluxML/NNlib.jl/pull/434 ).